web-ng/src/app/components/form-field-editor/form-field-editor.component.ts

```diff
@@ -97,10 +106,18 @@ export class FormFieldEditorComponent implements OnInit, OnChanges {
     });
   }
 
+  /*
+    Input params change from the parent are handled by this method.
+  */
   ngOnChanges(changes: SimpleChanges) {
-    const type = this.fieldTypes.find(
-      fieldType => fieldType.type === Number(this.type)
-    );
+    const type =
```

Comment:
This is better, but I think a switch statement would be clearer here; find+lambda is overkill considering there are only two items that would potentially match. Also, the fact that "type" the local var refers to something different than "type" the class member is confusing. I'd simplify as follows:

1. Rename class member "type" to "fieldType".
2. Extract switch block into a helper function and refer to it directly when calling setValue below.

For ex:
```
this.formFieldGroup.setValue({
  label: this.label,
  required: this.required,
  type: this.getSelectedFieldTypeOption()
});
```

Then getSelectedFieldTypeOption() would be something like:
```
switch (this.type) {
  case FieldType.TEXT:
    return FormFieldEditor.FIELD_TYPE_OPTIONS[0];
  case FieldType.MULTIPLE:
    return FormFieldEditor.FIELD_TYPE_OPTIONS[this.cardinality === FieldCardinality.SINGLE ? 1 : 2];
  default:
    throw new Error(`Unsupported field type${this.type}`);
}
```

I'd also make "this.fieldTypes" into a private static constant "FIELD_TYPE_OPTIONS" to avoid further confusion.